### ISSUE TITLE
Reverse relations fetching

### ISSUE THREAD

**Issue by Forceres** [piccolo-orm/piccolo#1036]
FEATURE REQUEST:
Is it a problem to add reverse relation fetching (conceptual or technical)?
It would be nice to see something like that:
```python
   class Student(Table, tablename="students"):
       name = Varchar(length=64)
       teacher_id = ForeignKey(Teacher)

   class Teacher(Table, tablename="teachers"):
       name = Varchar(length=64)
       
   await Teacher.objects(Teacher.all_related())
```

**Comment by sinisaos** [comment]
@Forceres [Here](https://github.com/piccolo-orm/piccolo/pull/599) is an old reverse lookup PR that never took off. [Here](https://gist.github.com/sinisaos/5d9e7b36d42d6e125d93177a370dc7f4) is also workaround to get same result with existing code.

**Comment by Forceres** [comment]
Why old PR was never looked up?

**Comment by Forceres** [comment]
> @Forceres [Here](https://github.com/piccolo-orm/piccolo/pull/599) is an old reverse lookup PR that never took off. [Here](https://gist.github.com/sinisaos/5d9e7b36d42d6e125d93177a370dc7f4) is also workaround to get same result with existing code.

That workaround is too ineffective

**Comment by sinisaos** [comment]
@Forceres That PR was never properly reviewed by the Piccolo author. I don't remember what the reasons were for that (bad API design or something else).  Everything worked as expected at the time PR was made. Everything should work now also, but I'm not sure if it does after all the changes that have been made in Piccolo.

**Comment by Forceres** [comment]
> @Forceres That PR was never properly reviewed by the Piccolo author. I don't remember what the reasons were for that (bad API design or something else).  Everything worked as expected at the time PR was made. Everything should work now also, but I'm not sure if it does after all the changes that have been made in Piccolo.

I don't see any reasons for it to be declined, the only one solution with that PR is to use raw queries builder (written from zero to hero)

**Comment by sinisaos** [comment]
> I don't see any reasons for it to be declined, the only one solution with that PR is to use raw queries builder (written from zero to hero)

I agree with you, but you have to wait @dantownsend for an explanation. That PR is pretty similar to how the Piccolo [M2M](https://github.com/piccolo-orm/piccolo/blob/master/piccolo/columns/m2m.py) was built.

**Comment by Forceres** [comment]
> > I don't see any reasons for it to be declined, the only one solution with that PR is to use raw queries builder (written from zero to hero)
> 
> I agree with you, but you have to wait @dantownsend for an explanation. That PR is pretty similar to how the Piccolo [M2M](https://github.com/piccolo-orm/piccolo/blob/master/piccolo/columns/m2m.py) was built.

Someone has already tried to wait for him) But, anyway, there is no choice, forced to wait

**Comment by Forceres** [comment]
Modern ORM, which doesn't have basic functionality like reverse relations

**Comment by dantownsend** [comment]
I know it's frustrating when PRs don't get merged in, but it's just a matter of resources and time - I generally have to prioritise stuff I need for client work, as that's what pays my bills. Loads of my own PRs haven't been merged in either, because unless I'm 100% happy with it, and willing to support a feature in the future, I'm hesitant about merging it in.

I do come back to old PRs - so if they haven't been merged they're not 'dead'.

But I agree, this would be a very nice feature to have. I generally work around it with something like this:

```python
teacher = await Teacher.objects().get(name="Alice")
students = await Student.objects().where(Student.teacher == teacher)
```

**Comment by Forceres** [comment]
> I know it's frustrating when PRs don't get merged in, but it's just a matter of resources and time - I generally have to prioritise stuff I need for client work, as that's what pays my bills. Loads of my own PRs haven't been merged in either, because unless I'm 100% happy with it, and willing to support a feature in the future, I'm hesitant about merging it in.
> 
> I do come back to old PRs - so if they haven't been merged they're not 'dead'.
> 
> But I agree, this would be a very nice feature to have. I generally work around it with something like this:
> 
> ```python
> teacher = await Teacher.objects().get(name="Alice")
> students = await Student.objects().where(Student.teacher == teacher)
> ```

Got you. Hope you find time to resolve it.
That approach has one problem - it is 2 queries instead of one with joins

**Comment by Forceres** [comment]
> I know it's frustrating when PRs don't get merged in, but it's just a matter of resources and time - I generally have to prioritise stuff I need for client work, as that's what pays my bills. Loads of my own PRs haven't been merged in either, because unless I'm 100% happy with it, and willing to support a feature in the future, I'm hesitant about merging it in.
> 
> I do come back to old PRs - so if they haven't been merged they're not 'dead'.
> 
> But I agree, this would be a very nice feature to have. I generally work around it with something like this:
> 
> ```python
> teacher = await Teacher.objects().get(name="Alice")
> students = await Student.objects().where(Student.teacher == teacher)
> ```

Any news?
